lib/pages/general/item/item.dart

```diff
@@ -223,6 +229,65 @@ class ItemPage extends StatelessWidget {
       'UpgradeComponent',
       'Weapon'
     ].contains(item.type))  {
+      if (item.flags != null && item.flags.length > 0 && item.type != 'Gathering') {
```

Comment:
This was my original approach and the function I wrote was almost identical to this one, but I must have been doing it in the wrong  place as the compiler wouldn't allow me to compile the project with this function in it.

I'll re-implement it using this process, but in a different way.

Reply:
I've re-implemented this now using this approach. The original implementation (and this one) return the really helpful error message `type 'WhereIterable<String>' is not a subtype of type 'List<String>'`

To get around this, the `.where` has to be on an `Iterable<String>` not a `List<String>`, but of course this needs to be re-cast back to a `List<String>` after processing...

I've kept some of the `if` statements as well, as they were processing elements in the list based on whether or not _other_ keys were in the list at the same time, a test not possible in a `where` loop.

Reply:
> I've re-implemented this now using this approach. The original implementation (and this one) return the really helpful error message type 'WhereIterable<String>' is not a subtype of type 'List<String>'
You can fix that error by using .toList() after the .where() 😃

Reply:
Thinking more on this, the `.where` approach is an additive selection, building the new list based only on the explicitly desired keys, whereas the original implementation was a subtractive selection, _removing_ the unwanted keys and leaving the rest.

In the example above (the greatsword), the `.where` loop will only return the `Account Bound` state unless all the other keys are listed as well in the boolean check, so this may not actually be more efficient than the original set of `if` statements.

I've expanded the boolean filters by way of example.

Reply:
You could also reverse the check. Something along the lines of:
```
List<String> getFilteredFlags(List<String> originalFlags) {
    if (originalFlags == null) return []; // Return an empty list if the flags are null. Now you only have to check if there are more than zero flags to determine whether or not to run the _buildFlags function.
    return originalFlags
       .where((flag) {
               // Perform your checks here, something along the lines of:
               switch (section) {
                   case 'equipment':
                       if (flag == 'Unwanted' || flag == 'Unwanted2') return false;
                       return true;
                   case 'bank':
                       if (flag == 'Unwanted') return false;
                       return true;
                   // Etc.
                   default: return false;
       })
       .toList();
}
```

Reply:
> You can fix that error by using .toList() after the .where() 😃

Huh, this didn't occur to me even though I was doing that immediately after on the new variable... Most recent commit fixes this :)

Reply:
I've also rewritten the `.where` loop based on exclusion rather than inclusion.